pynbody/gravity/calc.py

```diff
@@ -83,14 +83,7 @@ def midplane_rot_curve(f, rxy_points, eps=None, mode=config['gravity_calculation
             "OpenMP module is now selected at install time", DeprecationWarning)
 
     if eps is None:
-        try:
-            eps = f['eps']
-        except KeyError:
-            eps = np.zeros(len(f))
-            eps += f.properties['eps']
-
-    if isinstance(eps, str):
```

Comment:
For backwards compatibility I think we need to allow eps to be a string that gets auto-converted into a unit. This doesn't seem to be possible in the new version.

Reply:
The prerequisite of the function `direct` in _gravity.pyx is that eps should be a SimArray of the same dtype as `f['mass']` and the same units as `f['pos']`. For that before every call to `direct` I think it is good to make sure `eps` is a SimArray.